couchdb/server/state.py

```diff
@@ -2,11 +2,37 @@
 #
 import logging
 
-__all__ = ('add_fun', 'add_lib')
+__all__ = ('add_fun', 'add_lib', 'reset')
 
 log = logging.getLogger(__name__)
 
 
+def reset(server, config=None):
+    """Resets query server state.
+
+    :command: reset
+
+    :param server: Query server instance.
+    :type server: :class:`~couchdb.server.BaseQueryServer`
+
+    :param config: Optional dict argument to set up query config.
+    :type config: dict
+
+    :return: True
+    :rtype: bool
+    """
+    log.debug('Reset server state')
+    del server.state['functions'][:]
+    del server.state['functions_src'][:]
+    server.state['query_config'].clear()
+    if config is not None:
+        log.debug('Set new query config:\n%s', config)
+        server.state['query_config'].update(config)
+    if server.version >= (1, 1, 0):
+        server.state['view_lib'] = ''
+    return True
```

Comment:
That's an artefact from JS query server backport. [It has](https://github.com/apache/couchdb/blob/master/share/server/views.js#L36-L37) line_length value in state, but we eventually don't use it. Instead we [have](https://github.com/djc/couchdb-python/pull/286/commits/d00497dad989463d762334aee0c4c73c7df17d00#diff-01f1fb5540ab6f33c873acc84ad1143bR104) it used differently without using state.